### ISSUE TITLE
Exception in render-ws-example

### ISSUE THREAD

**Issue by unidesigner** [saalfeldlab/render#96]
Hi!

I try to work through the examples `docs/src/site/markdown/render-ws-example.md` but get an error calling:
```
${CLIENT_SCRIPTS}/manage_stacks.sh ${EXAMPLE_1_PARAMS} --stack v1_acquire --action SET_STATE --stackState COMPLETE

  Running: /home/stephan/image/alignment/render/deploy/jdk1.8.0_131/bin/java -cp /home/stephan/image/alignment/render/render-ws-java-client/target/render-ws-java-client-2.0.3-SNAPSHOT-standalone.jar -Xms1G -Xmx1G -Djava.awt.headless=true -XX:+UseSerialGC org.janelia.render.client.StackClient --baseDataUrl http://localhost:8080/render-ws/v1 --owner demo --project example_1 --stack v1_acquire --action SET_STATE --stackState COMPLETE


23:31:50.135 [main] INFO  [org.janelia.render.client.ClientRunner] run: entry
23:31:50.319 [main] INFO  [org.janelia.render.client.StackClient] runClient: entry, parameters={
  "renderWeb" : {
    "baseDataUrl" : "http://localhost:8080/render-ws/v1",
    "owner" : "demo",
    "project" : "example_1"
  },
  "stack" : "v1_acquire",
  "action" : "SET_STATE",
  "stackState" : "COMPLETE"
}
23:31:50.630 [main] INFO  [org.janelia.render.client.RenderDataClient] getStackMetaData: submitting GET http://localhost:8080/render-ws/v1/owner/demo/project/example_1/stack/v1_acquire
23:31:50.730 [main] INFO  [org.janelia.render.client.StackClient] setStackState: before update, stackMetaData={
  "stackId" : {
    "owner" : "demo",
    "project" : "example_1",
    "stack" : "v1_acquire"
  },
  "state" : "LOADING",
  "lastModifiedTimestamp" : "2019-02-11T22:28:19.247Z",
  "currentVersionNumber" : 0,
  "currentVersion" : {
    "createTimestamp" : "2019-02-11T22:28:19.151Z",
    "cycleNumber" : 1,
    "cycleStepNumber" : 1
  }
}
23:31:50.746 [main] INFO  [org.janelia.render.client.RenderDataClient] setStackState: submitting PUT http://localhost:8080/render-ws/v1/owner/demo/project/example_1/stack/v1_acquire/state/COMPLETE
23:31:50.771 [main] ERROR [org.janelia.render.client.ClientRunner] run: caught exception
org.apache.http.client.ClientProtocolException: HTTP status 500 with body

  Command failed with error 9: 'The 'cursor' option is required, except for aggregate with the explain argument' on server localhost:27017. The full response is { "ok" : 0.0, "errmsg" : "The 'cursor' option is required, except for aggregate with the explain argument", "code" : 9, "codeName" : "FailedToParse" }

returned for

  PUT http://localhost:8080/render-ws/v1/owner/demo/project/example_1/stack/v1_acquire/state/COMPLETE

	at org.janelia.render.client.response.BaseResponseHandler.getValidatedResponseEntity(BaseResponseHandler.java:94) ~[render-ws-java-client-2.0.3-SNAPSHOT-standalone.jar:na]
	at org.janelia.render.client.response.ResourceCreatedResponseHandler.handleResponse(ResourceCreatedResponseHandler.java:30) ~[render-ws-java-client-2.0.3-SNAPSHOT-standalone.jar:na]
	at org.janelia.render.client.response.ResourceCreatedResponseHandler.handleResponse(ResourceCreatedResponseHandler.java:15) ~[render-ws-java-client-2.0.3-SNAPSHOT-standalone.jar:na]
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:218) ~[render-ws-java-client-2.0.3-SNAPSHOT-standalone.jar:na]
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:160) ~[render-ws-java-client-2.0.3-SNAPSHOT-standalone.jar:na]
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:136) ~[render-ws-java-client-2.0.3-SNAPSHOT-standalone.jar:na]
	at org.janelia.render.client.RenderDataClient.setStackState(RenderDataClient.java:571) ~[render-ws-java-client-2.0.3-SNAPSHOT-standalone.jar:na]
	at org.janelia.render.client.StackClient.setStackState(StackClient.java:291) [render-ws-java-client-2.0.3-SNAPSHOT-standalone.jar:na]
	at org.janelia.render.client.StackClient$1.runClient(StackClient.java:179) ~[render-ws-java-client-2.0.3-SNAPSHOT-standalone.jar:na]
	at org.janelia.render.client.ClientRunner.run(ClientRunner.java:38) ~[render-ws-java-client-2.0.3-SNAPSHOT-standalone.jar:na]
	at org.janelia.render.client.StackClient.main(StackClient.java:187) [render-ws-java-client-2.0.3-SNAPSHOT-standalone.jar:na]
23:31:50.771 [main] INFO  [org.janelia.render.client.ClientRunner] run: exit, processing failed after 0 hours, 0 minutes, 0 seconds
```

**Comment by trautmane** [comment]
Hi Stephan,

I think this might be a mongodb version issue.  
Do you know which version you are using?
Render currently works with mongodb 3.2 - I'm guessing you might be running with 3.6.

Best,
Eric

**Comment by unidesigner** [comment]
Hi Eric,

Indeed, I'm running mongodb 4.0.5. Is there a possibility to run it with this version? The previous example calls seemed to work fine. I'm not sure if it is possible to install/run mongodb 3.2 in parallel.

Thanks,
Stephan

**Comment by trautmane** [comment]
Hi Stephan,

I've updated the code base to use a newer mongodb java driver that will work with 4.0.*.  
I verified that it fixes the problem you found and that it is backward compatible since we are still using 3.2 at Janelia.  If you want to try it right away, you'll need to pull it from the hierarchical branch (see commit above).  I'm planning to merge the hierarchical branch to the master branch soon to create a render 2.1.0 release, but I need to confirm that is okay with folks that are using render at the Allen Institute first.

If you are just playing around, another option is to set up a mongo 3.2 instance in docker.

Best,
Eric
